### DISCUSSION TITLE
Can you use a variable in topic name?

### DISCUSSION THREAD

**Discussion by MJohnson459** [foxglove/community#8] [Q&A] [answered]
I'm trying out foxglove and I've created some plots for a robot, `robot_1` with topics like `/robot_1/cmd_vel.angular.z` and `/robot_1/cmd_vel.linear.x`, etc. If I have a second robot `robot_2`, is there a way to switch all the plots over to point to it? The only way I can see at the moment is manually changing it for each topic in each plot.

I see there are variables in foxglove but I don't see a reference for if they can be used in topic names?

**Comment by jtbandes** [comment, marked as answer]
No, this isn't possible currently. Variables can be only be used in filters. See existing ticket: https://github.com/foxglove/studio/issues/3284

**Reply by jtbandes** [reply]
Also related: https://github.com/foxglove/studio/issues/2907
